src/main/java/be/seeseemelk/mockbukkit/entity/PlayerMock.java

```diff
@@ -959,18 +964,28 @@ public void setSneaking(boolean sneaking)
 		this.sneaking = sneaking;
 	}
 
+	public void simulateSneak(boolean sneak)
+	{
+		PlayerToggleSneakEvent event = new PlayerToggleSneakEvent(this, sneak);
+		Bukkit.getPluginManager().callEvent(event);
+	}
+
 	@Override
 	public boolean isSprinting()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return sprinting;
 	}
 
 	@Override
 	public void setSprinting(boolean sprinting)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		this.sprinting = sprinting;
+	}
+
+	public void simulateSprint(boolean sprint)
+	{
+		PlayerToggleSprintEvent event = new PlayerToggleSprintEvent(this, sprint);
+		Bukkit.getPluginManager().callEvent(event);
```

Comment:
The method should respect the Event outcome and set the `sprinting` state accordingly.